src/Scripting/CoreTest/ScriptOptionsTests.cs

```diff
@@ -179,5 +180,83 @@ public void WithFileEncoding_SameValueTwice_DoesNotCreateNewInstance()
             var options = ScriptOptions.Default.WithFileEncoding(Encoding.ASCII);
             Assert.Same(options, options.WithFileEncoding(Encoding.ASCII));
         }
+
+        [Fact]
+        public void WithAllowUnsafe_SetsAllowUnsafe()
+        {
+            Assert.True(ScriptOptions.Default.WithAllowUnsafe(true).AllowUnsafe);
```

Comment:
>Assert.True [](start = 12, length = 11)

`Assert.Same(ScriptOptions.Default.WithAllowUnsafe(true))` would be better since we want to test that when the option does not change it's the same instance